src/main/common/clients/launchDarklyClient.ts

```diff
@@ -0,0 +1,32 @@
+import * as config from 'config'
+import { User } from 'idam/user'
+import * as ld from 'ldclient-node'
+
+const sdkKeys: string = config.get<string>('launchDarkly.sdk')
+
+export class LaunchDarklyClient {
+  private static client: ld.LDClient
+
+  constructor () {
+    LaunchDarklyClient.initClient()
+  }
+
+  static initClient () {
+    if (!LaunchDarklyClient.client) {
+      LaunchDarklyClient.client = ld.init(sdkKeys)
+    }
+  }
+
+  async callFeatureFlag (user: User, featureKey: string, f?: () => void) {
+    const ldUser: ld.LDUser = {
+      key: user.id
```

Comment:
I think you need to pass the user roles here as well

Reply:
I have added it to LDUser as custom attributes